File: csharp/Chapter09/NegativeBalanceException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace com.db4o.dg2db4o.chapter9
{
    class NegativeBalanceException : Exception
    {
        double _balance;

        public NegativeBalanceException(double balance)
        {
            _balance = balance;
        }

        public override string ToString()
        {   
            return "NegativeBalanceException (Balance = " + System.String.Format("{0:C}",_balance) + ")";
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter9/NegativeBalanceException.java

package com.db4o.dg2db4o.chapter9;

import java.text.NumberFormat;

public class NegativeBalanceException extends Exception
{
    double _balance;

        public NegativeBalanceException(double balance)
        {
            _balance = balance;
        }

        public String toString()
        {   
            return "NegativeBalanceException (Balance = " + NumberFormat.getCurrencyInstance().format(_balance) + ")";
        }
}
